File: backend/models/business_account.py
# backend/models/business_account.py

from sqlalchemy import Column, String, ForeignKey, JSON
from sqlalchemy.orm import relationship
from .base import Base

class BusinessAccount(Base):
    __tablename__ = "business_accounts"

    id = Column(String, primary_key=True, index=True)  # e.g., BIZ-0001
    name = Column(String, unique=True, index=True)
    industry_id = Column(String, ForeignKey("industries.id"))
    owner_id = Column(String, index=True)  # Optional: Can link to user table later
    features_enabled = Column(JSON, nullable=True)  # e.g., { "loyalty": true, "billing": true }

    industry = relationship("Industry", backref="business_accounts")

File: frontend/lib/models/admin/business_account.dart
// lib/models/admin/business_account.dart

class BusinessAccount {
  final String id;
  final String name;
  final String industryId;
  final String? ownerId;
  final Map<String, dynamic>? featuresEnabled;

  BusinessAccount({
    required this.id,
    required this.name,
    required this.industryId,
    this.ownerId,
    this.featuresEnabled,
  });

  factory BusinessAccount.fromJson(Map<String, dynamic> json) {
    return BusinessAccount(
      id: json['id'],
      name: json['name'],
      industryId: json['industry_id'],
      ownerId: json['owner_id'],
      featuresEnabled: json['features_enabled'] ?? {},
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'industry_id': industryId,
      'owner_id': ownerId,
      'features_enabled': featuresEnabled,
    };
  }
}
